packages/rpc-graphql/src/loaders/account.ts

```diff
@@ -1,47 +1,319 @@
+                });
+
+                // Now place the orphans by searching for the best fetch with
+                // encoding provided
+                if (
+                    addedHashes.length !== 0 &&
+                    addedHashes.some(
+                        hash =>
+                            accountFetchesByArgsHash[hash].args.encoding &&
+                            accountFetchesByArgsHash[hash].args.encoding !== 'base58',
+                    )
+                ) {
+                    // At least one account fetch specified encoding, so use that encoding for all
+                    // orphaned fetches
+                    for (const hash of addedHashes) {
+                        const entry = accountFetchesByArgsHash[hash];
+                        const encoding = entry.args.encoding;
+                        // `base58` should be avoided if possible, since it will cause errors
+                        // from the RPC for larger accounts
+                        if (encoding && encoding !== 'base58') {
+                            orphanedFetches.forEach(({ promiseCallback }) => {
+                                entry.addresses[address].callbacks.push({
+                                    callback: promiseCallback,
+                                    dataSlice: entry.args.dataSlice ?? null,
+                                });
+                            });
+                            return;
+                        }
+                    }
+                } else {
+                    // No account fetches specified encoding, so use `base64`
+                    orphanedFetches.forEach(({ args: orphanedArgs, promiseCallback: orphanPromiseCallback }) => {
+                        const { commitment, dataSlice, minContextSlot } = orphanedArgs;
+                        const defaultOrphanArgs: AccountLoaderArgsBase = {
+                            commitment,
+                            dataSlice,
+                            encoding: 'base64',
+                            minContextSlot,
+                        };
+                        addFetchByArgsHash(defaultOrphanArgs, {
+                            callback: orphanPromiseCallback,
+                            dataSlice: defaultOrphanArgs.dataSlice ?? null,
+                        });
+                    });
+                }
+            });
+
+            /**
+             * For each set of accounts related to some common args, fetch them in the fewest number
+             * of network requests.
+             */
+            Object.values(accountFetchesByArgsHash).forEach(async ({ args, addresses: addressCallbacks }) => {
+                const addresses = Object.keys(addressCallbacks) as Address[];
+                if (addresses.length === 1) {
+                    const address = addresses[0];
+                    try {
+                        const result = await resolveAccountUsingRpc({ address, ...args });
```

Comment:
Same problem as below. This is an `await` in a loop – the loop started here:

```ts
Object.values(accountFetchesByArgsHash).forEach(...);
```

You want to set up all the promises and launch them at once.

```ts
await Promise.all(
    Object.values(accountFetchesByArgsHash).map(...),
);
```